execution/native/account.go

```diff
@@ -0,0 +1,218 @@
+package native
+
+import (
+	"bytes"
+
+	"github.com/hyperledger/burrow/acm"
+	"github.com/hyperledger/burrow/acm/acmstate"
+	"github.com/hyperledger/burrow/crypto"
+	"github.com/hyperledger/burrow/crypto/sha3"
+	"github.com/hyperledger/burrow/deploy/compile"
+	"github.com/hyperledger/burrow/execution/errors"
+	"github.com/hyperledger/burrow/txs/payload"
+)
+
+func CreateAccount(st acmstate.ReaderWriter, address crypto.Address) error {
+	acc, err := st.GetAccount(address)
+	if err != nil {
+		return err
+	}
+	if acc != nil {
+		if acc.NativeName != "" {
+			return errors.ErrorCodef(errors.ErrorCodeReservedAddress,
+				"cannot create account at %v because that address is reserved for a native contract '%s'",
+				address, acc.NativeName)
+		}
+		return errors.ErrorCodef(errors.ErrorCodeDuplicateAddress,
+			"tried to create an account at an address that already exists: %v", address)
+	}
+	return st.UpdateAccount(&acm.Account{Address: address})
+}
+
+func InitCode(st acmstate.ReaderWriter, address crypto.Address, code []byte) error {
+	return initCode(st, address, nil, code)
+}
+
+func InitChildCode(st acmstate.ReaderWriter, address crypto.Address, parent crypto.Address, code []byte) error {
+	return initCode(st, address, &parent, code)
+}
+
+func initCode(st acmstate.ReaderWriter, address crypto.Address, parent *crypto.Address, code []byte) error {
+	acc, err := mustAccount(st, address)
+	if err != nil {
+		return err
+	}
+	if acc.EVMCode != nil || acc.WASMCode != nil {
+		return errors.ErrorCodef(errors.ErrorCodeIllegalWrite,
+			"tried to initialise code for a contract that already has code: %v", address)
+	}
+
+	acc.EVMCode = code
+
+	// keccak256 hash of a contract's code
+	hash := sha3.NewKeccak256()
```

Comment:
Not necessarily something to be addressed in this PR, but I'm unsure why we maintain our own sha3 library in `crypto` when we can use the standard library? See: https://github.com/hyperledger/burrow/blob/master/crypto/hash.go#L11

Reply:
I had no idea it existed in crypto - looks like it was committed 5th May 2018 so I guess that's why! I will change.